internal/librarian/generate.go

```diff
@@ -125,25 +125,33 @@ func (r *generateRunner) run(ctx context.Context) error {
 	if err := os.Mkdir(outputDir, 0755); err != nil {
 		return fmt.Errorf("failed to make output directory, %s: %w", outputDir, err)
 	}
-	slog.Info("Code will be generated", "dir", outputDir)
-
-	additionalMsg := ""
+	// The last generated commit is changed after library generation,
+	// use this map to keep the mapping from library id to commit sha before the
+	// generation since we need these commits to create pull request body.
+	idToCommits := make(map[string]string, 0)
+	failedLibraries := make([]string, 0)
```

Comment:
nit: slices never need to make made prefer

```suggestion
	var failedLibraries []string
```